src/dmclock_server.h

```diff
@@ -989,10 +983,10 @@ namespace crimson {
 	for (auto& r : client.requests) {
 	  r.tag.reservation -= client.info->reservation_inv;
 
-#ifndef DO_NOT_DELAY_TAG_CALC
-	  // reduce only for front tag. because next tags' value are invalid
-	  break;
-#endif
+	  if (tag_calc == TagCalc::Delayed) {
```

Comment:
Would it be better to flip the nexting (word?) of the r-loop and conditional, so we don't run the conditional each time through the loop? So out the outer level, if we're delayed, we simply adjust the reservation for the first request. But if we're immediate only then we execute the loop and adjust all reservation tags.